types/testing-library__jest-dom/index.d.ts

```diff
@@ -0,0 +1,36 @@
+// Type definitions for @testing-library/jest-dom 5.0
+// Project: https://github.com/testing-library/jest-dom
+// Definitions by: Ernesto García <https://github.com/gnapse>
+//                 John Gozde <https://github.com/jgoz>
+//                 Seth Macpherson <https://github.com/smacpherson64>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+/// <reference types="jest" />
+
+declare namespace jest {
+    interface Matchers<R, T> {
+        /**
+         * @deprecated
+         */
+        toBeInTheDOM(container?: HTMLElement | SVGElement): R;
+        toBeInTheDocument(): R;
+        toBeVisible(): R;
+        toBeEmpty(): R;
+        toBeDisabled(): R;
+        toBeEnabled(): R;
+        toBeInvalid(): R;
+        toBeRequired(): R;
+        toBeValid(): R;
+        toContainElement(element: HTMLElement | SVGElement | null): R;
+        toContainHTML(htmlText: string): R;
+        toHaveAttribute(attr: string, value?: any): R;
+        toHaveClass(...classNames: string[]): R;
+        toHaveFocus(): R;
+        toHaveFormValues(expectedValues: { [name: string]: any }): R;
```

Comment:
```suggestion
        toHaveFormValues(expectedValues: Record<string, unknown>): R;
```

Since this targets typescript 3.0 we can use `unknown` instead. `{ [name: string]: any }` has a special behavior in that it allows any object type. But this actually does not consider Symbol keys in the implementation. It should throw for those case as well as when passing e.g. functions. It's a niche edge case but for new types we might as well do it right.

See https://github.com/microsoft/TypeScript/wiki/Breaking-Changes#-k-string-unknown--is-no-longer-a-wildcard-assignment-target